usage-dash-oauth.py

```diff
@@ -0,0 +1,102 @@
+import dash
+from dash.dependencies import Input, Output, State
+import dash_core_components as dcc
+import dash_html_components as html
+
+import binascii
+import json
+import flask
+import os
+
+from dash_auth.dash_oauth import DashTokenAuth
+
+
+server = flask.Flask(__name__)
+server.secret_key = (
+    'fd8e7047cd8c3bedef10568f124aad5102899537a5ec49e15af1d195caf4'
+)
+USERS = [
+    {
+        'username': 'chris',
+        'password': 'snowstorm',
+        'secret-token': binascii.hexlify(os.urandom(24)),
+        'has-access': True
+    },
+    {
+        'username': 'tom',
+        'password': 'rainfall',
+        'secret-token': binascii.hexlify(os.urandom(24)),
+        'has-access': False
+    }
+]
+
+class DashAuth(DashTokenAuth):
+    def __init__(self, app):
+        DashTokenAuth.__init__(self, app)
+
+    def validate_user(self, username, password):
+        print('validate_user')
+        user = [u for u in USERS if u['username'] == username]
+        if len(user) == 0:
+            return (False, '',)
+        else:
+            if user[0]['password'] == password and user[0]['has-access']:
+                return (True, user[0]['secret-token'],)
+            else:
+                return (False, '',)
+
+    def validate_token(self, user_token):
```

Comment:
`validate_token` checks if the user, identified by their `user_token`, has access to the app. The `user_token` is stored as a cookie in the user's browser after they log in. This function is called after the user's "5 minutes of access" is expired to check that they still have access to the app. By identifying the user's session with this unique, secret user token, we can continuously check if the user still has access to the app without requiring them to explicitly log in again.